### ISSUE TITLE
Assertion failed: scales_input.is_weights()

### ISSUE THREAD

**Issue by ghylander** [NVIDIA/TensorRT#1346]
## Description
The full error meesage is the following:

ERROR: builtin_op_importers.cpp:3460 In function importUpsample:
[8] Assertion failed: scales_input.is_weights()

the error comes out after a few layers have been translated

this fails with a model with the following (deviation from working exports):

- Input shape is (None, None, 3)
- there is a tf.keras.layers.experimental.preprocessing.Resizing() layer
- there is a tf.keras.layers.experimental.preprocessing.CenterCrop() layer

i have tried to fold the model using onnx graphsurgeon and polygraphy, while the folding is succeful the export result is the same

## Environment

**TensorRT Version**: 7.1.3
**NVIDIA GPU**: jetson nano
**NVIDIA Driver Version**: n/a
**CUDA Version**: 10.2
**CUDNN Version**: 8.0
**Operating System**: ubuntu 18.04
**Python Version (if applicable)**: 2.8.5
**Tensorflow Version (if applicable)**: 2.5

## Steps To Reproduce

/usr/src/tensorrt/bin/trtexec --onnx={onnxFilePath} --fp16 --workspace=1024 --warmUp=2 --dumpProfile --verbose --explicitBatch --saveEngine={engineFilePath}

**Comment by pranavm-nvidia** [comment]
Would you be able to share the original (non-folded) ONNX model? 

Also, which versions of Polygraphy, ONNX-GraphSurgeon and ONNX are you using? If you could post the output of `python3 -m pip freeze | grep -e "onnx\|polygraphy"` that'd be helpful.

**Comment by ghylander** [comment]
Hi, i'll put this on hold as i have spotted a bug with saving models in tensorflow 2.5 in the pipeline, before this step
i reverted to tensorflow 2.4 and will attempt the process again

**Comment by ghylander** [comment]
hi, please find the onnx model here: https://drive.google.com/file/d/1uieXzRvmbzS1TCVaV24Hs17oJXx2ESV9/view?usp=sharing
the issue with tf was resolved and the onnx->trt happens both when exporting from tf 2.4.2 and tf 2.5

this model is untrained, but the export issue is the exact same

**Comment by ghylander** [comment]
forgot to attach the output of onnx and polygraphy package versions:

```
onnx==1.9.0
onnx-graphsurgeon==0.3.9
onnxruntime==1.8.0
polygraphy==0.29.2
tf2onnx==1.8.5
```

**Comment by pranavm-nvidia** [comment]
Looks like this is the problematic part of the graph:
![resize_subgraph](https://user-images.githubusercontent.com/49246958/125654169-0abb875b-2927-4788-93b6-d276046d7e8c.png)

There are a few different things you could try:
- Export the model with a newer opset. The current model is using 9, and if you switch to 13 for example, the structure might be more friendly to TRT. 
- Try using a newer version of TRT, like TRT 8. 
- In the worst case, you can simply remove the resize/upsample portion of the graph and do this as a preprocessing step instead. AFAICT, all it's doing is resizing the input to `[1, 345, 796, 3]`. To edit the ONNX graph, you can use Polygraphy:
    ```bash
    polygraphy surgeon extract cc_untrained.onnx \
        --inputs StatefulPartitionedCall/model_1/resizing_1/resize/ResizeBilinear:0:["batch","height","width",3]:float32 \
        -o cc_untrained_no_resize.onnx
    ```

    `cc_untrained_no_resize.onnx` will look like this:

    ![no_resize](https://user-images.githubusercontent.com/49246958/125654869-012e867e-8134-4add-b171-21ff20f13784.png)

**Comment by ghylander** [comment]
hi and thanks for the help
first i tried to export to onnx using opset 14 and 13, but the export to trt fails with a different error message (last few lines of the output) (both opset 13 and 14 produced the same error): 

```
[07/15/2021-09:40:58] [V] [TRT] ModelImporter.cpp:125: StatefulPartitionedCall/model_1/MobilenetV3large/expanded_conv_3/squeeze_excite/AvgPool/Mean_Squeeze__1403 [Squeeze] inputs: [StatefulPartitionedCall/model_1/MobilenetV3large/expanded_conv_3/squeeze_excite/AvgPool/Mean:0 -> (-1, 72, 1, 1)], [const_axes__1402 -> (2)],
terminate called after throwing an instance of 'std::out_of_range'
  what():  Attribute not found: axes
Aborted (core dumped)
```

then, i tried removing the tf.keras.layers.experimental.preprocessing.Resizing() and performing the resize outside the network, this introduced a new error:

```
[07/15/2021-09:56:59] [V] [TRT] ModelImporter.cpp:90: Importing initializer: ConstantFolding/StatefulPartitionedCall/model/center_crop/truediv_recip:0
[07/15/2021-09:56:59] [E] [TRT] onnx2trt_utils.cpp:467: Found unsupported datatype (11) when importing initializer: ConstantFolding/StatefulPartitionedCall/model/center_crop/truediv_recip:0
ERROR: ModelImporter.cpp:92 In function parseGraph:
[8] Assertion failed: convertOnnxWeights(initializer, &weights, ctx)
[07/15/2021-09:56:59] [E] Failed to parse onnx file
[07/15/2021-09:56:59] [E] Parsing model failed
[07/15/2021-09:56:59] [E] Engine creation failed
[07/15/2021-09:56:59] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec # /usr/src/tensorrt/bin/trtexec --onnx=cc_untrained.onnx --fp16 --workspace=1024 --warmUp=2 --dumpProfile --verbose --explicitBatch --saveEngine=cc_untrained.engine
```
I'll try updating TRT next, but it'll take a bit of testing to see it doesn't break anything in the pipeline

**Comment by ghylander** [comment]
have a look at these two models, as we could drop the upsample part and it's only doing a bilinear resizing now:
this one has both Resize and CenterCrop layers: https://drive.google.com/file/d/1ZgSUuyUT23qC0mDTF6AZRZtAEiIFv0Bt/view?usp=sharing
This one only has CenterCrop: https://drive.google.com/file/d/1Qt6czLRlDPr3QkToGwlWnzTQjpVpLASK/view?usp=sharing

First link fails with the first output in the previous post, second link fails with the output in the second log above

**Comment by ghylander** [comment]
It seems to be a problem with the CenterCrop, as i changed the input shape to a specified one instead of None, None and i still get the same "attribute not found: axes" issue

**Comment by ghylander** [comment]
the axes issue seems to be related to the opset, as the exporting onnx -> TRT operation fails for onnx models using opset 14 but not opset9

**Comment by pranavm-nvidia** [comment]
@ghylander TRT 7.1.3 only [supports up to opset 11](https://github.com/onnx/onnx-tensorrt/blob/7.1/operators.md), so that's probably why you're seeing those errors. So you'd either need to export using opset 11 or upgrade to TRT 8

**Comment by ghylander** [comment]
that's quite bad because latest supported version of TRT on jetson nano is 7.1.3

**Comment by ghylander** [comment]
Nevertheless, i still run into issues if when creating the model with tensorflow i specify the inputs to be of shape (None, None, 3)
This happens with opset9 and otherwise working models
the error i get is:

```
[07/19/2021-10:41:45] [E] [TRT] onnx2trt_utils.cpp:467: Found unsupported datatype (11) when importing initializer: ConstantFolding/StatefulPartitionedCall/model/center_crop/truediv_recip:0
ERROR: ModelImporter.cpp:92 In function parseGraph:
[8] Assertion failed: convertOnnxWeights(initializer, &weights, ctx)
[07/19/2021-10:41:45] [E] Failed to parse onnx file
[07/19/2021-10:41:45] [E] Parsing model failed
[07/19/2021-10:41:45] [E] Engine creation failed
[07/19/2021-10:41:45] [E] Engine set up failed
```

**Comment by pranavm-nvidia** [comment]
In that case I'd suggest using the workaround [I mentioned earlier](https://github.com/NVIDIA/TensorRT/issues/1346#issuecomment-880020882):

> In the worst case, you can simply remove the resize/upsample portion of the graph and do this as a preprocessing step instead. AFAICT, all it's doing is resizing the input to [1, 345, 796, 3]. To edit the ONNX graph, you can use Polygraphy:
```     
polygraphy surgeon extract cc_untrained.onnx \
    --inputs StatefulPartitionedCall/model_1/resizing_1/resize/ResizeBilinear:0:["batch","height","width",3]:float32 \
    -o cc_untrained_no_resize.onnx
```

**Comment by ghylander** [comment]
i can simply not include the layer when building the model in tensorflow, although i wanted to include the resize and crop operations as layers to simplify deployment and make the model a bit more plug and play
let's hope TRT is updated soon for the jetson platform

**Comment by ghylander** [comment]
i will close this until TRT 8 is supported on jetson nano
if after updating TRT and using onnx opset 13+ there are still issues, i will reopen it
thanks for the support
